scarlet/frame.py

```diff
@@ -45,11 +45,12 @@ def __init__(
                 self._psfs = PSF(psfs)
 
         assert channels is None or len(channels) == self.shape[0]
+        if channels is None:
+            logger.warning(
+                "No channels specified. Possible, but multi-observation processing will likely fail!")
```

Comment:
I don't like this: if it is likely to fail, we should disallow it. Is there an expert use case for `channels=None`?

Reply:
I'm fine with that. I did make it so that `bbox` can no longer be `None` because the code before would have actually failed if sources didn't have them. In this case it won't fail with no `channels` unless the user has mutliple observations with non-overlapping bands , so I left open the possibility of creating a `Frame` without channels. But I'm fine to also make that field required.

Reply:
It would certainly be best practice to set the channel labels. But I've used multi-observation cases without that, so I don't think the statement is strictly true that it will likely fail.

I see these options to be valid: 
1. require channels to be set
2. check that multiple observations work without setting channels for any of them

My problem is that the warning itself isn't helpful. It either works or it doesn't. We leave the user to guess.

Reply:
Ok, at first I didn't understand how this ever worked and after looking into this I don't think that it worked as expected. This is the code before Remy's last PR to match the model frame to an observation frame: https://github.com/pmelchior/scarlet/blob/1b6635155bbdd38457819699cd5e9dbb128fd394/scarlet/observation.py#L91-L102 .

So it "worked" when channels was `None` in both the model and observation frame by implicitly assuming that they overlapped on the first channel. So if, for example, a model was initialized with 5 bands and an observation overlapped on bands 4 and 5, then when calling `Observation.get_loss` it would be updating the wrong bands in the model.

The above code block changed in Remy's PR when he modified the `Frame` class to use a different `__and__` method, which still requires channels but only works when comparing two frames. It provided inconsistent behavior if it was comparing a `Frame` and a `Box`, which is undesirable, and is incompatible with the simplified slicing algorithm.

So I don't _think_ that it ever worked with channels that didn't overlap unless you initialized `Frame` with a `Box` that specifically set the origin of the channel dimension correctly, and that only worked because the model PSF is the same in all bands.

So the question is do we _ever_ allow users to not set the channels, knowing that it can't work in multiband mode and might lead to unexpected results? I am leaning toward "no" unless either of you can think of a good reason why we should allow this.

Reply:
I agree with you, we should disallow it. If a pattern is so prone to failure (as it is because even we can't do it correctly), it should be banned. This might explain the color shifts we can see [here](https://pmelchior.github.io/scarlet/tutorials/multiresolution.html#View-Full-Model).

Reply:
I don't think that is causing those color shifts because that notebook defines the channels and they are matched properly. But I'll see what this notebook looks like on my branch tomorrow.

Reply:
Color shifts are still there, so this didn't fix that.

Reply:
I wrote the previous channel matching code. And it did work *under assumptions*, namely that someone construction the model frame properly, i.e. with a concatenation of the channels. I think it is better to be explicit about it even if that causes a minor hassle for the single-observation case.

I'm ok with it now, but it'll surely come up again when we debate the general `Observation` API and background mechanics.